xonsh/base_shell.py

```diff
@@ -38,12 +117,25 @@ def default(self, line):
         code = self.push(line)
         if code is None:
             return
+        hist = builtins.__xonsh_history__
+        ts1 = None
+        tee = Tee() if builtins.__xonsh_env__.get('XONSH_STORE_STDOUT', False) \
+                    else io.StringIO()
         try:
+            ts0 = time.time()
             self.execer.exec(code, mode='single', glbs=self.ctx)  # no locals
+            ts1 = time.time()
+            hist.last_cmd_rtn = 0  # returncode for success
```

Comment:
If we start a process and it fails, it looks like this will still come up 0 ("success") since that exception isn't raised up, right?